Add DynamicConfigProvider for Schema Registry (#11362)
apache/druid: docs/ingestion/data-formats.md

| url | String | Specifies the url endpoint of the Schema Registry. | yes |
| capacity | Integer | Specifies the max size of the cache (default = Integer.MAX_VALUE). | no |
| urls | Array<String> | Specifies the url endpoints of the multiple Schema Registry instances. | yes(if `url` is not provided) |

> | config | Json | To send additional configurations, configured for Schema Registry.  User can implement a `DynamicConfigProvider` to supply some properties at runtime, by adding `"druid.dynamic.config.provider"`:`{"type": "<registered_dynamic_config_provider_name>", ...}` in json. | no |

Reviewer: I suggest linking to `operations/dynamic-config-provider.md` instead of providing an example inline, maybe something like 
```suggestion
| config | Json | To send additional configurations, configured for Schema Registry. This can be supplied via a [DynamicConfigProvider](../../operations/dynamic-config-provider.md). | no |
```
or something similar (and for the other sections that have been updated)

Author: Good suggestion! I'll change it.
